sdk/azcore/internal/exported/pipeline.go

```diff
@@ -57,21 +60,35 @@ func (tp transportPolicy) Do(req *Request) (*http.Response, error) {
 
 // NewPipeline creates a new Pipeline object from the specified Policies.
 // Not directly exported, but used as part of runtime.NewPipeline().
-func NewPipeline(transport Transporter, policies ...Policy) Pipeline {
+func NewPipeline(tr tracing.Tracer, transport Transporter, policies ...Policy) Pipeline {
 	// transport policy must always be the last in the slice
 	policies = append(policies, transportPolicy{trans: transport})
 	return Pipeline{
 		policies: policies,
+		tracer:   tr,
 	}
 }
 
+// SetTracer sets the provided [tracing.Tracer] on [Pipeline].
+// Not directly exported, but used when creating/cloning clients.
+func SetTracer(pl *Pipeline, tracer tracing.Tracer) {
+	pl.tracer = tracer
+}
+
```

Comment:
Pipelines are supposed to be immutable and this makes them mutable. I see that you say "used when crating/cloning clients" so, if this is used correctly then this shouldn't be a problem, but it may not always be used correctly. Also (assuming we keep this) , why wouldn't this be a method with a *Pipeline receiver? 
In addition. this can fix the tracer problem but what about the context capturing the raw response or having custom headers or some other feature - this context should also not flow to the credential's pipeline. I think we need a fix for this problem still and ideally one solution could fix all these issues.